### PR TITLE
impl-notification-calls

### PR DESCRIPTION
**Pull request by Velrok**
Based on #13 . Merge that first.

### CODE REVIEW

**Comment by Velrok** [comment]
Based on #13 . Merge that first.

File: wobble/__init__.py [hunk 1]
```diff
@@ -11,19 +11,28 @@ class WobbleService(object):
...
-        return wrapper
+            return fn(self, *args)
+        return requires_login_decorator
 
     def log_calls(fn):
-        def wrapper(self, *args):
-            result = fn(self, args)
+        def log_calls_decorator(*params):
+            self = params[0]
+            args = params[1:]
```
Comment (zeisss) [inline wobble/__init__.py:23]: What is, if there are no additional parameters?

File: wobble/__init__.py [hunk 2]
```diff
@@ -34,14 +43,22 @@ def __init__(self, api_endpoint='http://wobble.moinz.de/api/endpoint.php',
...
+    def __enter__(self):
+        return self
+
+    def __exit__(self, exception=None, error=None, traceback=None):
+        if exception:
+            logging.error(exception)
+            logging.error(error)
+            logging.error(traceback)
+
+        self.user_signout()
```
Comment (zeisss) [inline wobble/__init__.py:55]: What if the `login()` fails? Then there is no session to logout ...

Comment (Velrok) [inline wobble/__init__.py:55]: Good question I will investigate.

**Pull request merged**